src/test/java/com/teragrep/pth_06/planner/walker/conditions/EarliestConditionTest.java

```diff
@@ -22,4 +24,24 @@ void conditionTest(final Document document) {
         Condition elementCondition = new EarliestCondition(element).condition();
         Assertions.assertEquals(e, elementCondition.toString());
     }
+
+    @Test
+    @ExtendWith(DocumentExtension.class)
+    void equalityTest(final Document document) {
+        Element element = document.createElement("test");
+        element.setAttribute("value", "946677600");
+        element.setAttribute("operation", "EQUALS");
+        Element anotherElement = document.createElement("test");
+        anotherElement.setAttribute("value", "1000");
+        anotherElement.setAttribute("operation", "EQUALS");
+        EarliestCondition expected = new EarliestCondition(element);
+        EarliestCondition actual = new EarliestCondition(element);
+        actual.condition();
+        EarliestCondition notExpected = new EarliestCondition(anotherElement);
+        Assertions.assertTrue(new QueryCondition.Smart().compare(expected.condition(), actual.condition()));
+        Assertions.assertFalse(new QueryCondition.Smart().compare(expected.condition(), notExpected.condition()));
```

Comment:
why not override `equals(Object o);` for each other condition classes?

Reply:
that was indeed the better solution I implemented that